packages/itwinui-react/src/core/Tabs/Tabs.tsx

```diff
@@ -230,6 +146,75 @@ export const Tabs = (props: TabsProps) => {
     }
   }, [currentActiveIndex, type, orientation, tabsWidth]);
 
+  return (
+    <Box
+      className={cx('iui-tabs-wrapper', `iui-${orientation}`, className)}
+      style={stripeProperties}
+      ref={refs}
+      {...rest}
+    >
+      <TabsContext.Provider
+        value={{
+          orientation,
+          type,
+          color,
+          activeIndex,
+          focusActivationMode,
+          currentActiveIndex,
+          setCurrentActiveIndex,
+          onTabSelected,
+          overflowOptions,
+        }}
+      >
+        {children}
+      </TabsContext.Provider>
+    </Box>
+  );
+}) as PolymorphicForwardRefComponent<'div', TabsComponentOwnProps>;
+TabsComponent.displayName = 'Tabs';
+
+// ----------------------------------------------------------------------------
+// Tabs.TabList component
+
+type TabsTabListOwnProps = {
+  /**
+   * Tab items.
+   */
+  children: React.ReactNode[] | React.ReactNode;
+};
+
+const TabsTabList = React.forwardRef((props, ref) => {
+  const { className, children, ...rest } = props;
+
+  const items = Array.isArray(children) ? children : [children];
+
+  const tablistRef = React.useRef<HTMLUListElement>(null);
+  const refs = useMergedRefs(tablistRef, ref);
+
+  const isClient = useIsClient();
+
+  const {
+    orientation,
+    type,
+    color,
+    activeIndex,
+    focusActivationMode,
+    currentActiveIndex,
+    setCurrentActiveIndex,
+    onTabSelected,
+    overflowOptions,
+  } = useSafeContext(TabsContext);
+
+  useIsomorphicLayoutEffect(() => {
+    if (
+      activeIndex != null &&
+      currentActiveIndex !== activeIndex &&
+      setCurrentActiveIndex
+    ) {
+      setCurrentActiveIndex(getBoundedValue(activeIndex, 0, items.length - 1));
+    }
+  }, [activeIndex, currentActiveIndex, items.length]);
+
   const [focusedIndex, setFocusedIndex] = React.useState<number | undefined>();
   React.useEffect(() => {
     if (tablistRef.current && focusedIndex !== undefined) {
```

Comment:
This hack needs to be addressed in this PR. Any suggestions on how to fix this hack would be appreciated

Reply:
This hack has been addressed by using `tablistRef` and `focusedIndex` to locate the correct tab to focus.